src/types/capsule.rs

```diff
@@ -164,77 +167,113 @@ impl PyCapsule {
     /// }
     ///
     /// Python::with_gil(|py| {
-    ///     let name = CString::new("foo").unwrap();
     ///     let capsule =
-    ///         PyCapsule::new_with_destructor(py, 123, &name, destructor as fn(u32, *mut c_void))
+    ///         PyCapsule::new_with_destructor(py, 123, None, destructor as fn(u32, *mut c_void))
     ///             .unwrap();
     ///     let context = Box::new(tx);  // `Sender<String>` is our context, box it up and ship it!
-    ///     capsule.set_context(py, Box::into_raw(context) as *mut c_void).unwrap();
+    ///     capsule.set_context(Box::into_raw(context) as *mut c_void).unwrap();
     ///     // This scope will end, causing our destructor to be called...
     /// });
     ///
     /// assert_eq!(rx.recv(), Ok("Destructor called!".to_string()));
     /// ```
     #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    pub fn set_context(&self, py: Python<'_>, context: *mut c_void) -> PyResult<()> {
+    pub fn set_context(&self, context: *mut c_void) -> PyResult<()> {
         let result = unsafe { ffi::PyCapsule_SetContext(self.as_ptr(), context) as u8 };
         if result != 0 {
-            Err(PyErr::fetch(py))
+            Err(PyErr::fetch(self.py()))
         } else {
             Ok(())
         }
     }
 
     /// Gets the current context stored in the capsule. If there is no context, the pointer
     /// will be null.
-    pub fn get_context(&self, py: Python<'_>) -> PyResult<*mut c_void> {
+    ///
+    /// Returns an error if this capsule is not valid.
+    pub fn get_context(&self) -> PyResult<*mut c_void> {
```

Comment:
Since we are making a breaking change in any case, shouldn't this be named `context` following general Rust conventions as well as `name` and `pointer`?

Reply:
Pushed 👍